File: src/数组、链表、跳表/Linked List/206.反转链表.go
/*
 * @lc app=leetcode.cn id=206 lang=golang
 *
 * [206] 反转链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	res := reverseList(head.Next)
	head.Next.Next = head
	head.Next = nil
	return res
}
// func reverseList(head *ListNode) *ListNode {
// 	var prev *ListNode
// 	curr  := head
// 	for curr != nil {
// 		next := curr.Next
// 		curr.Next = prev
// 		prev = curr
// 		curr = next
// 	}
// 	return prev
// }
// @lc code=end

File: src/数组、链表、跳表/Linked List/206.反转链表.cs
/*
 * @lc app=leetcode.cn id=206 lang=csharp
 *
 * [206] 反转链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 public class Solution {
    public ListNode ReverseList(ListNode head) {
        if(head == null || head.next == null){
            return head;
        }
        var res = ReverseList(head.next);
        head.next.next = head;
        head.next = null;
        return res;
    }
}
// public class Solution {
//     public ListNode ReverseList(ListNode head) {
//         ListNode prev = null;
//         var curr = head;
//         while(curr != null) {
//             var next = curr.next;
//             curr.next = prev;
//             prev = curr;
//             curr = next;
//         }
//         return prev;
//     }
// }
// @lc code=end

